Core/Src/main.c

```diff
@@ -117,7 +148,147 @@ int main(void)
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)temp_val, 1, 100);
+	}
+	HAL_GPIO_WritePin(GPIOB, GPIO_PIN_12, GPIO_PIN_SET);
+
+	//correct temp data
+	adc_T = (temp_val[2] >> 24)|(temp_val[1] >> 21)|temp_val[0];
+
+	//read data
+	HAL_GPIO_WritePin(GPIOB, GPIO_PIN_12, GPIO_PIN_RESET);
+	HAL_GPIO_WritePin(GPIOB, GPIO_PIN_10, GPIO_PIN_SET);
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_T1_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digT1, 2, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digT1, 2, 100);
+	}
+
+	dig_T1 = (digT1[1] << 8) | digT1[0];
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_T2_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digT2, 2, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digT2, 2, 100);
+	}
+
+	dig_T2 = (int)(digT2[1] << 8) | (int) digT2[0];
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_T3_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digT3, 2, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digT3, 2, 100);
+	}
+
+	dig_T3 = (int) ((digT3[1] << 8) | digT3[0]);
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_H1_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (unsigned char *)&dig_H1, 1, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (unsigned char *)&dig_H1, 1, 100);
+	}
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_H2_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digH2, 2, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)digH2, 2, 100);
+	}
+
+	dig_H2 = (int) ((digH2[1] << 8) | digH2[0]);
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_H3_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (unsigned char *)&dig_H3, 1, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (unsigned char *)&dig_H3, 1, 100);
+	}
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_H4_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (uint8_t *)dig_H4_ls, 2, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)dig_H4_ls, 2, 100);
+	}
+
+	dig_H4_ls[1] = dig_H4_ls[1] & 0b0000000000001111;
+	dig_H4 = (int) (dig_H4_ls[0] | dig_H4_ls[1]);
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_H5_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (uint8_t *)dig_H5_ls, 2, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (uint8_t *)dig_H5_ls, 2, 100);
+	}
+
+	dig_H5_ls[0] = (dig_H5_ls[0] & 0b1111000000000000) >> 12;
+	dig_H5_ls[1] = dig_H5_ls[1] << 8;
+	dig_H5 = (int) (dig_H5_ls[1] | dig_H5_ls[0]);
+
+	HAL_SPI_Transmit(&hspi2, (uint8_t *)&dig_H6_addr, 1, 100);
+	ret = HAL_SPI_Receive(&hspi2, (unsigned char *)&digH6, 1, 100);
+	while(ret != HAL_OK){
+		ret = HAL_SPI_Receive(&hspi2, (unsigned char *)&digH6, 1, 100);
+	}
+
+	dig_H6 = (char) digH6;
+
+	HAL_GPIO_WritePin(GPIOB, GPIO_PIN_12, GPIO_PIN_SET);
+
+	//perform calculations
+	BME280_S32_t var1, var2;
+	var1 = ((((adc_T>>3) - ((BME280_S32_t)dig_T1<<1))) * ((BME280_S32_t)dig_T2)) >> 11;

```

Comment:
I don't understand what's going on here. I think encapsulating this part of the code into functions would be much better. 
If I want a function to read the humidity, I would suggest something like: 
```
float readHumidity()
{ 
    // Code to get humidity
}
```

In the header file you add the function prototype and in the source file you add the function definition.